flink-end-to-end-tests/flink-e2e-test-utils/src/main/java/org/apache/flink/streaming/tests/util/s3/S3UtilProgram.java

```diff
@@ -0,0 +1,221 @@
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * S3 utilities.
+ *
+ * <p>Usage: java -jar S3UtilProgram.jar args.
+ *
+ * <p>Note: {@code S3UtilProgram.Action.lineNumber*} actions are applicable only
+ * to valid non-compressed CSV comma separated files.
+ *
+ * <p>Program parameters:
+ * <ul>
+ *     <li>action (string, required): Action to perform, see {@link S3UtilProgram.Action}.</li>
+ *     <li>bucket (string, required): Bucket where s3 objects reside.</li>
+ *     <li>s3file (string, required for single object actions): s3 object key.</li>
+ *     <li>s3prefix (string, required for actions over objects grouped by key prefix): s3 key prefix.</li>
+ *     <li>s3filePrefix (string, optional for downloadByPrefix or numberOfLinesByPrefix): s3 file name prefix w/o directory to filter files by name.</li>
+ *     <li>localFile (string, required for single file actions): local file path.</li>
+ *     <li>localFolder (string, required for actions over folders): local folder path.</li>
+ *     <li>parallelism (int, default 10): parallelism for parallelizable actions (e.g. {@code numberOfLinesByPrefix}).</li>
+ * </ul>
+ */
+class S3UtilProgram {
+	enum Action {
+		listByPrefix,
+		downloadFile,
+		downloadByPrefix,
+		deleteFile,
+		deleteByPrefix,
+		numberOfLinesInFile,
+		numberOfLinesByPrefix
+	}
+
+	private static final Map<Action, Consumer<ParameterTool>> handlers;
+	static {
+		Map<Action, Consumer<ParameterTool>> handlersMutable = new HashMap<>();
+		handlersMutable.put(Action.listByPrefix, S3UtilProgram::listByPrefix);
+		handlersMutable.put(Action.downloadFile, S3UtilProgram::downloadFile);
+		handlersMutable.put(Action.downloadByPrefix, S3UtilProgram::downloadByPrefix);
+		handlersMutable.put(Action.deleteFile, S3UtilProgram::deleteFile);
+		handlersMutable.put(Action.deleteByPrefix, S3UtilProgram::deleteByPrefix);
+		handlersMutable.put(Action.numberOfLinesInFile, S3UtilProgram::numberOfLinesInFile);
+		handlersMutable.put(Action.numberOfLinesByPrefix, S3UtilProgram::numberOfLinesByPrefix);
+		handlers = Collections.unmodifiableMap(handlersMutable);
+	}
+
+	private static final String countQuery = "select count(*) from s3object";
+
+	public static void main(String[] args) {
+		final ParameterTool params = ParameterTool.fromArgs(args);
+		final Action action = Action.valueOf(params.getRequired("action"));
+		handlers.get(action).accept(params);
+	}
+
+	private static void listByPrefix(ParameterTool params) {
```

Comment:
Rename to `listByFilenamePrefix`

Reply:
this lists by full s3 object key prefix, could be then:
listByFullPrefix